File: Week6/Problem/sentimental-readability/readability.py
# TODO
from cs50 import get_string
import re
# Define the main function


def main():
    # Grab the text
    text = get_string("Text: ")

    # Call the function
    result = coleman(text)

    # Test the results
    if result < 1:
        print("Before Grade 1")
    elif result > 16:
        print("Grade 16+")
    else:
        print(f"Grade {result}")
# Create the function to calculate


def coleman(text):
    # Identiy the values
    characters, words, sentences = 0, 1, 0
    # Start the loop
    for i in text:
        if i in [".", "!", "?"]:
            sentences += 1
        elif i == " ":
            words += 1
        elif i.isalpha():
            characters += 1

    # Define the values for the formula
    l = 100 * (characters / words)
    s = 100 * (sentences / words)

    # Complete the formula
    result = (0.0588 * l) - (0.296 * s) - 15.8

    # Retrieve the value
    return round(result)


main()

File: Week2/Problem/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    printf("%s\n", text);

    // Initialize the variables for the formula
    float L = (count_letters(text) / (float) count_words(text)) * 100;
    float S = (count_sentences(text) / (float) count_words(text)) * 100;

    // Calculate the index
    int index = (int) round(0.0588 * L - 0.296 * S - 15.8);

    // Check against range values
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Checking the amount of letters on the string
// By iterating through each character
// And checking if it's alpha
int count_letters(string text)
{
    int counter = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            counter++;
        }
    }
    return counter;
}

// Checking the amount of words on the string
// By iterating through each character
// And checking the spaces, then we add 1
int count_words(string text)
{
    int counter = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    return counter;
}

// Checking the amount of sentences on the string
// By iterating through each character
// And checking the special characters to account for sentences
int count_sentences(string text)
{
    int counter = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i + 1] == '.' || text[i + 1] == '!' || text[i + 1] == '?')
        {
            counter++;
        }
    }
    return counter;
}